packages/react-server/core/util/ClientCssHelper.js

```diff
@@ -29,49 +31,51 @@ module.exports = {
 			throw new Error("ClientCssHelper.registerPageLoad can't be called server-side");
 		}
 
-		var newCss = pageObject.getHeadStylesheets();
+		return Q.all(PageUtil.standardizeStyles(pageObject.getHeadStylesheets())).then(newCss => {
```

Comment:
Hmmm... good point.

Previously (and as written in this PR) the swap of "remove old and add new" is relatively atomic. The remove/add of the dom nodes happen in the same time slice (which I assume matters, but I haven't actually tested extensively, I don't think). I could imagine adding all the new styles as they come in, then removing the old ones all at once? Seems like that could have some unintentional side effects with styles though. 

Thoughts?